docs/tutorials/bucket_tutorial.jl

```diff
@@ -0,0 +1,315 @@
+# for the timestepping, and [DiffEqCallbacks.jl](https://github.com/SciML/DiffEqCallbacks.jl)
+# is used as described below,
+# for accessing the solver state during the integration.
+using OrdinaryDiffEq: ODEProblem, solve, Euler
+using DiffEqCallbacks
+# We use [ClimaCore](https://github.com/CliMA/ClimaCore.jl)
+# for setting up the domain/coordinate points. While
+# this infrastructure isn't really necessary for standalone simulations,
+# adhering to it makes setting up coupled simulations very easy. It also
+# is nice to rely on ClimaCore utilities because they have been designed
+# in advance for running distributed simulations.
+using ClimaCore
+
+# We also use CLIMAParameters, which strives to ensure a common set of
+# parameters across all Clima models, and to make parameter estimation
+# more seamless.
+import CLIMAParameters as CP
+
+# Lastly, let's bring in the bucket model types (from ClimaLSM) that we
+# will need access to.
+
+using ClimaLSM.Bucket:
+    BucketModel,
+    BucketModelParameters,
+    PrescribedAtmosphere,
+    PrescribedRadiativeFluxes,
+    BulkAlbedo
+using ClimaLSM.Domains: coordinates, Point
+using ClimaLSM: initialize, make_update_aux, make_ode_function
+# We also want to plot the solution
+using Plots
+
+FT = Float64;
+
+# As mentioned we use CLIMAParameters for earth parameters that are
+# required across models (e.g. the density of water and ice, the latent
+# heat of fusion at a reference temperature, etc). The land model requires
+# additional parameters as described in the text above. These two sets
+# are combined in the object `BucketModelParameters` as follows:
+import ClimaLSM
+include(joinpath(pkgdir(ClimaLSM), "parameters", "create_parameters.jl"));
+earth_param_set = create_lsm_parameters(FT);
+
+# Define our `BulkAlbedo` model using a constant soil and snow albedo:
+# The soil albedo is a function of your coordinates, which would be
+# (x,y) on a plane, and (lat,lon) on a sphere. In the future, we
+# will support other albedo models.
+α_soil = (coordinate_point) -> FT(0.2);
+α_snow = FT(0.8);
+albedo = BulkAlbedo{FT}(α_snow, α_soil);
+# The critical snow level setting the scale for when we interpolate between
+# snow and soil albedo
+S_c = FT(0.2);
+# The field capacity of the soil
+W_f = FT(0.15);
+# The soil depth and base temperature
+d_soil = FT(10.0);
+T_base = FT(280.0);
+# Roughness lengths (meters)
+z_0m = FT(1e-2);
+z_0b = FT(1e-3);
+# Thermal parameters of soil
+κ_soil = FT(0.0);
+ρc_soil = FT(2e6);
+bucket_parameters = BucketModelParameters(
+    d_soil,
+    T_base,
+    κ_soil,
+    ρc_soil,
+    albedo,
+    S_c,
+    W_f,
+    z_0m,
+    z_0b,
+    earth_param_set,
+);
+
+# Set up the model domain. At every coordinate point, we'll solve
+# an ODE for `W`, `Ws`, and `T_sfc`. In coupled simulations run at the same
+# resolution as the atmosphere, the bucket domain would match the
```

Comment:
I don't know if you'd like to include it here, but technically the coupled run should be able to deal with different resolutions too.